sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala

```diff
@@ -68,4 +68,444 @@ private[xml] trait TestXmlData {
+          |  </array>
+          |</arrayOfArray1>
+          |  <arrayOfArray2>
+          |  <array>
+          |    <item>
+          |      <inner1>str1</inner1>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |<arrayOfArray2>
+          |  <array/>
+          |  <array>
+          |    <item>
+          |      <inner2>str3</inner2>
+          |      <inner2>str33</inner2>
+          |    </item>
+          |    <item>
+          |      <inner2>str4</inner2>
+          |      <inner1>str11</inner1>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |<arrayOfArray2>
+          |  <array>
+          |    <item>
+          |      <inner3>
+          |        <inner4>2</inner4>
+          |        <inner4>3</inner4>
+          |      </inner3>
+          |      <inner3/>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |</ROW>
+          |""".stripMargin :: Nil
+
+  def nullsInArrays: Seq[String] =
+    """<ROW>
+          <field1>
+            <array>
+              <item/>
+            </array>
+            <array>
+              <array>
+                <array>
+                  <item>Test</item>
+                </array>
+              </array>
+            </array>
+          </field1>
+        </ROW>""" ::
+    """
+        <ROW>
+          <field2>
+            <item/>
+            <array>
+              <item>
+                <Test>1</Test>
+              </item>
+            </array>
+          </field2>
+        </ROW>""" ::
+    """
+        <ROW>
+          <field3>
+            <array>
+              <item/>
+            </array>
+            <array>
+              <item>
+                <Test>2</Test>
+              </item>
+            </array>
+          </field3>
+        </ROW>""" ::
+    """
+        <ROW>
+          <field4>
+            <array>
+              <item/>
+              <array>
+                <item>1</item>
+                <item>2</item>
+                <item>3</item>
+              </array>
+            </array>
+          </field4>
+        </ROW>""" :: Nil
+
+  def corruptRecords: Seq[String] =
+    """<ROW>""" ::
+    """""" ::
+    """<ROW>
+          |  <a>1</a>
+          |  <b>2</b>
+          |</ROW>""".stripMargin ::
+    """
+          |<ROW>
+          |  <a>str_a_4</a>
+          |  <b>str_b_4</b>
+          |  <c>str_c_4</c>
+          |</ROW>
+          |""".stripMargin ::
+    """
+          |</ROW>
+          |""".stripMargin :: Nil
+
+  def emptyRecords: Seq[String] =
+    """<ROW>
+          <a><struct></struct></a>
+        </ROW>""" ::
+    """<ROW>
+          <a>
+            <struct><b><c/></b></struct>
+          </a>
+        </ROW>""" ::
+    """<ROW>
+          <b>
+            <item>
+              <c><struct></struct></c>
+            </item>
+            <item/>
+          </b>
+        </ROW>""" :: Nil
+
+  def arrayAndStructRecords: Seq[String] =
```

Comment:
unused

Reply:
I plan to add tests in a follow-up PR. I will remove this for clarity in this PR.